code/crates/engine/src/host.rs

```diff
@@ -8,33 +8,64 @@ use malachitebft_core_consensus::{Role, VoteExtensionError};
 use malachitebft_core_types::{CommitCertificate, Context, Round, ValueId, VoteExtensions};
 use malachitebft_sync::{PeerId, RawDecidedValue};
 
-use crate::consensus::ConsensusRef;
 use crate::util::streaming::StreamMessage;
 
 pub use malachitebft_core_consensus::{LocallyProposedValue, ProposedValue};
 
 /// A reference to the host actor.
 pub type HostRef<Ctx> = ActorRef<HostMsg<Ctx>>;
 
+/// What to do next after a decision.
+#[derive_where(Debug)]
+pub enum Next<Ctx: Context> {
+    /// Start at the given height with the given validator set.
+    Start(Ctx::Height, Ctx::ValidatorSet),
+
+    /// Restart at the given height with the given validator set.
+    Restart(Ctx::Height, Ctx::ValidatorSet),
+}
+
 /// Messages that need to be handled by the host actor.
 #[derive_where(Debug)]
 pub enum HostMsg<Ctx: Context> {
-    /// Consensus is ready
-    ConsensusReady(ConsensusRef<Ctx>),
+    /// Notifies the application that consensus is ready.
+    ///
+    /// The application MUST reply with a message to instruct
+    /// consensus to start at a given height.
+    ConsensusReady {
+        /// Use this reply port to instruct consensus to start the first height.
+        reply_to: RpcReplyPort<(Ctx::Height, Ctx::ValidatorSet)>,
```

Comment:
Would the interface be more homogeneous if we used here `Next::Start`?
Although I think we can't do that without opening the door to "accepting" `Next::Restart` at build time, even if we add a sanity check at runtime.
Maybe something to consider later on if we manage to remove the `Restart` variant in the future.